node/services/service.go

```diff
@@ -0,0 +1,36 @@
+package service
+
+import (
+	"context"
+
+	"github.com/celestiaorg/celestia-node/service/block"
+
+	ipld "github.com/ipfs/go-ipld-format"
+	"go.uber.org/fx"
+)
+
+// Components wraps services creation
+func Components() fx.Option {
+	return fx.Options(
+		fx.Provide(func(params blockServiceParams) *block.Service {
+			service := block.NewBlockService(params.Fetcher, params.Store)
+			params.Lc.Append(fx.Hook{
+				OnStart: func(ctx context.Context) error {
+					return service.Start(ctx)
+				},
+				OnStop: func(ctx context.Context) error {
+					return service.Stop(ctx)
+				},
```

Comment:
@vgonkivs, Now, instead of wrapping Stop/Start with anon func, you can pass those funds directly to OnStart/OnStop

Reply:
@Wondertan , thanks for support. Passed start/stop directly to hook.